code/game/machinery/recycler.dm

```diff
@@ -104,8 +104,8 @@
 
 	var/items_recycled = 0
 
-	for(var/i in to_eat)
-		var/atom/movable/AM = i
+	for(var/i = to_eat.len, i > 0, i--) // Reverse order so it eats deepest contents before their containers. The opposite goes... badly.
+		var/atom/movable/AM = to_eat[i]

```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#forvara-in-list-vs-forvari-in-1-to-listlen

Reply:
wonder if replacing 101 with `reverseRange(AM0.GetAllContents(), start=1, end=0)` then checking for QDELETED objects would fix it, I think you will still run into the issue of QDELETED/ING items being in the list though so you should check for those.

Reply:
The problem is that GetAllContents returns a list starting with the container and going inwards, so the innermost items are last. It's in the exact wrong order that it needs to be in.

Going through it in reverse order solves it perfectly, because it deletes innermost contents first and it'll only ever end up trying to eat a QDEL'd item if deleting a certain item inside of something deletes the container, which i'm not aware of any examples.

Performance wise, this should be fine. I tested it under pretty extreme scenarios and it was fine. Combined with the conveyor belt maximum move limit, there should be no issue unless you create a elaborate setup where you force a LOT of items into it at once with disposals  or something. I'll test it.